charmcraft/application/commands/lifecycle.py

```diff
@@ -173,13 +173,46 @@ def run_managed(self, parsed_args: argparse.Namespace) -> bool:
         # notably here, the bundle plugin doesn't work on Windows.
         if sys.platform == "linux" and charmcraft_yaml and charmcraft_yaml.get("type") == "bundle":
             return False
+
         return super().run_managed(parsed_args)
 
+    def _update_charm_libs(self) -> None:
+        """Update charm libs attached to the project."""
+        craft_cli.emit.progress("Checking that charmlibs match 'charmcraft.yaml' values")
+        project = cast(models.CharmcraftProject, self._services.project)
+        libs_svc = cast(services.CharmLibsService, self._services.charm_libs)
+        installable_libs: list[models.CharmLib] = []
+        for lib in project.charm_libs:
+            charm_name, _, lib_name = lib.lib.partition(".")
```

Comment:
what is the library? `<charm-name>.<lib-name>` or just `<lib-name>`, `project.charm_libs` implies the former, `get_lib_version` the latter. I would probably encapsulate the details of splitting (as it seems technical) to the service itself

Reply:
The convention (easiest shown in the [LibData class](https://github.com/canonical/charmcraft/blob/main/charmcraft/utils/charmlibs.py#L33-L44)) doesn't really have a name for what `<charm-name>.<lib-name>` is - `full_name` is being used for `charms.<charm-name>.v<api-version>.<lib-name>`.